_posts/2020-12-04-release-1.12.0.md

```diff
@@ -0,0 +1,332 @@
+
+### Kubernetes High Availability (HA) Service
+
+Kubernetes provides built-in functionalities that Flink can leverage for JobManager failover, instead of relying on [ZooKeeper](https://zookeeper.apache.org/). To enable a “ZooKeeperless” HA setup, the community implemented a Kubernetes HA service in Flink 1.12 ([FLIP-144](https://cwiki.apache.org/confluence/x/H0V4CQ)). The service is built on the same [base interface]({{ site.DOCS_BASE_URL }}flink-docs-release-1.12/api/java/org/apache/flink/runtime/highavailability/HighAvailabilityServices.html) as the ZooKeeper implementation and uses Kubernetes’ [ConfigMap](https://kubernetes.io/docs/concepts/configuration/configmap/) objects to handle all the metadata needed to recover from a JobManager failure. For more details and examples on how to configure a highly available Kubernetes cluster, check out the [documentation]({{ site.DOCS_BASE_URL }}flink-docs-release-1.12/deployment/ha/kubernetes_ha.html).
+
+<div class="alert alert-info small" markdown="1">
+<b>Note:</b> This does not mean that the ZooKeeper dependency will be dropped, just that there will be an alternative for users of Flink on Kubernetes.
+</div>
+
+<hr>
+
+### Other Improvements
+
+**Migration of existing connectors to the new Data Source API**
+
+The previous release introduced a new Data Source API ([FLIP-27](https://cwiki.apache.org/confluence/display/FLINK/FLIP-27%3A+Refactor+Source+Interface)), allowing to implement connectors that work both as bounded (batch) and unbounded (streaming) sources. In Flink 1.12, the community started porting existing source connectors to the new interfaces, starting with the FileSystem connector ([FLINK-19161](https://issues.apache.org/jira/browse/FLINK-19161)). 
+
+<div class="alert alert-danger small" markdown="1">
+<b>Attention:</b> The unified source implementations will be completely separate connectors that are not snapshot-compatible with their legacy counterparts.
+</div>
+
+**Pipelined Region Scheduling ([FLIP-119](https://cwiki.apache.org/confluence/display/FLINK/FLIP-119+Pipelined+Region+Scheduling#FLIP119PipelinedRegionScheduling-BulkSlotAllocation))**
+
+Flink’s scheduler has been largely designed to address batch and streaming workloads separately. This release introduces a **unified** scheduling strategy that identifies sets of tasks connected via blocking data exchanges to break down the execution graph into _pipelined regions_. This allows to schedule each region only when there’s data to perform work and only deploy it once all the required resources are available; as well as to restart failed regions independently. In particular for batch jobs, the new strategy leads to more efficient resource utilization and eliminates deadlocks.
```

Comment:
> connected via blocking data exchanges

I think it should be "connected via pipelined data exchanges"

Reply:
I think "blocking" is a more common term here, and is aligned with the terminology in [this blogpost](https://github.com/apache/flink-web/pull/398/files#) by @azagrebin coming out after the release.

Reply:
Data exchange can be either pipelined or blocking. Pipelined regions are actually pipelined connected regions.
Also from the blogpost, there is "A _pipelined region_ is a subset of _subtasks_ in the _ExecutionGraph_ connected by _[pipelined](#intermediate-results)_ data exchanges."